app/src/common/Validations.js

```diff
@@ -0,0 +1,80 @@
+class Validations {
+
+    static typeOfRegexp(content, type) {
+        switch (type) {
+            case 'name':
+                return Validations.regexpName(content);
+            case 'email':
+                return Validations.regexpEmail(content);
+            case 'gender':
+                return Validations.regexpGender(content);
+            case 'age':
+                return Validations.regExpAge(content);
+        }
+    }
+
+    static validateForm(props) {
+        for (let i in props) {
+            if (props[i] !== props.middleName && !props[i].isValid) {
+                return false;
+            }
+        }
+
+        if (!props.middleName.isValid && !props.middleName.content) {
+            return true;
+        } else if (!props.middleName.isValid && props.middleName.content) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    static regexpImage(img) {
+        const imgPath = img.value;
+        const validImageExtensions = /(\.jpg|\.jpeg|\.png|\.bmp)$/i;
+        const minSize = 40000;
+        const maxSize = 5120000;
+        if (!validImageExtensions.exec(imgPath)) {
+            console.log('bad file');
+            return false;
+        } else {
+            if (img.files && img.files[0]) {
+                if (img.files[0].size >= minSize && img.files[0].size <= maxSize) {
+                    return true;
+                } else {
+                    console.log('invalid size');
```

Comment:
in the case when invalid size you can return false. So this if...else statement have to be refactored.